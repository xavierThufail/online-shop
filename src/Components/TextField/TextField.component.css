.ant-row .ant-form-item {
  margin-bottom: 0px;
}