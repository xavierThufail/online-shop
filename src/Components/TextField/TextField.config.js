import PropTypes from 'prop-types';

const propTypes = {
  style: PropTypes.object,
  name: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  placeholder: PropTypes.string,
  maxLength: PropTypes.number
};

const defaultProps = {
  label: '',
  style: {},
  placeholder: ''
};

const displayName = 'TextField';

export default {
  propTypes,
  defaultProps,
  displayName
};
