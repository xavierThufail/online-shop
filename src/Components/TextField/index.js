export { default as TextField } from './TextField.component';
