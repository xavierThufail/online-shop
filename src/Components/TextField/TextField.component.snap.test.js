import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, act, fireEvent } from '@testing-library/react';
import { Formik } from 'formik';

import TextField from './TextField.component';
import '../../matchMedia.mock';

describe('TextField component snap test', () => {

  beforeEach(jest.clearAllMocks);

  it('Render TextField, should matching snapshot', () => {
    const mockedProps = {
      name: 'test'
    };
    
    const { container } = render(
      <Formik>
        {formikProps => <TextField {...formikProps} {...mockedProps} />}
      </Formik>
    );

    expect(container).toMatchSnapshot();
  });

  it('Render TextField with label, should matching snapshot', () => {
    const mockedProps = {
      label: 'Username',
      name: 'username',
      placeholder: 'username'
    };

    const { container, getByText, getByPlaceholderText } = render(
      <Formik>
        {formikProps => <TextField {...formikProps} {...mockedProps} />}
      </Formik>  
    );
    const label = getByText(mockedProps.label);
    const input = getByPlaceholderText(mockedProps.placeholder);

    act(() => {
      fireEvent.blur(input);
    });

    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
