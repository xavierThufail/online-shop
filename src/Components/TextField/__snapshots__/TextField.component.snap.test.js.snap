// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component snap test Render TextField with label, should matching snapshot 1`] = `
<div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class="ant-form-item-no-colon"
        title="Username"
      >
        Username
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            class="ant-input"
            name="username"
            placeholder="username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-form-item-explain"
      >
        <div
          role="alert"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TextField component snap test Render TextField, should matching snapshot 1`] = `
<div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            class="ant-input"
            name="test"
            placeholder=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-form-item-explain"
      >
        <div
          role="alert"
        />
      </div>
    </div>
  </div>
</div>
`;
